syntax = "proto3";
package pt.ulisboa.tecnico.nameserver.contract;


message RegisterRequest {
  string name = 1;
  string qualifier = 2;
  string target = 3;
}

message RegisterResponse {
  string success = 1;
}

message LookupRequest {
  string name = 1;
  string qualifier = 2;
}

message LookupResponse {
  repeated string servers = 1;
}

message DeleteRequest {
  string name = 1;
  string target = 2;
}

message DeleteResponse {
  string success = 1;
}

service NamingServerService {
  rpc register(RegisterRequest) returns (RegisterResponse);
  rpc lookup(LookupRequest) returns (LookupResponse);
  rpc delete(DeleteRequest) returns (DeleteResponse);
}

